src/cascadia/UnitTests_TerminalCore/ScreenSizeLimitsTest.cpp

```diff
@@ -0,0 +1,136 @@
+using namespace Microsoft::Terminal::Core;
+
+namespace TerminalCoreUnitTests
+{
+class MockTermSettings : public winrt::implements<MockTermSettings, ICoreSettings>
+{
+public:
+    MockTermSettings(int32_t historySize, int32_t initialRows, int32_t initialCols) :
+        _historySize(historySize),
+        _initialRows(initialRows),
+        _initialCols(initialCols)
+    { }
+
+    // property getters - all implemented
+    int32_t HistorySize() { return _historySize; }
+    int32_t InitialRows() { return _initialRows; }
+    int32_t InitialCols() { return _initialCols; }
+    uint32_t DefaultForeground() { return COLOR_WHITE; }
+    uint32_t DefaultBackground() { return COLOR_BLACK; }
+    bool SnapOnInput() { return false; }
+    uint32_t CursorColor() { return COLOR_WHITE; }
+    CursorStyle CursorShape() const noexcept { return CursorStyle::Vintage; }
+    uint32_t CursorHeight() { return 42UL; }
+
+    // other implemented methods
+    uint32_t GetColorTableEntry(int32_t) const { return 123; }
+
+    // property setters - all unimplemented
+    void HistorySize(int32_t) { }
+    void InitialRows(int32_t) { }
+    void InitialCols(int32_t) { }
+    void DefaultForeground(uint32_t) { }
+    void DefaultBackground(uint32_t) { }
+    void SnapOnInput(bool) { }
+    void CursorColor(uint32_t) { }
+    void CursorShape(CursorStyle const&) noexcept { }
+    void CursorHeight(uint32_t) { }
+
+    // other unimplemented methods
+    void SetColorTableEntry(int32_t /* index */, uint32_t /* value */) { }
+
+private:
+    int32_t _historySize;
+    int32_t _initialRows;
+    int32_t _initialCols;
+};
+
+#define WCS(x) WCSHELPER(x)
+#define WCSHELPER(x) L#x
+
+class ScreenSizeLimitsTest
+{
+    TEST_CLASS(ScreenSizeLimitsTest);
+
+    TEST_METHOD(ScreenWidthAndHeightAreClampedToBounds)
+    {
+        DummyRenderTarget emptyRenderTarget;
+
+        // Negative values for initial visible row count or column count
+        // are clamped to 1. Too-large positive values are clamped to SHRT_MAX.
+        auto negativeColumnsSettings = winrt::make<MockTermSettings>(10000, 9999999, -1234);
+        Terminal negativeColumnsTerminal;
+        negativeColumnsTerminal.CreateFromSettings(negativeColumnsSettings, emptyRenderTarget);
+        COORD actualDimensions = negativeColumnsTerminal.GetViewport().Dimensions();
+        VERIFY_ARE_EQUAL(actualDimensions.Y, SHRT_MAX, L"Row count clamped to SHRT_MAX == " WCS(SHRT_MAX));
+        VERIFY_ARE_EQUAL(actualDimensions.X, 1, L"Column count clamped to 1");
+
+        // Zero values are clamped to 1 as well.
+        auto zeroRowsSettings = winrt::make<MockTermSettings>(10000, 0, 9999999);
+        Terminal zeroRowsTerminal;
+        zeroRowsTerminal.CreateFromSettings(zeroRowsSettings, emptyRenderTarget);
+        actualDimensions = zeroRowsTerminal.GetViewport().Dimensions();
+        VERIFY_ARE_EQUAL(actualDimensions.Y, 1, L"Row count clamped to 1" WCS(SHRT_MAX));

```

Comment:
```suggestion
        VERIFY_ARE_EQUAL(actualDimensions.Y, 1, L"Row count clamped to 1");
```

The `SHRT_MAX` here doesn't sound right.

Reply:
Good catch, fixed